pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/utils/SegmentMetadataUtils.java

```diff
@@ -50,11 +51,15 @@ public static void savePropertiesConfiguration(PropertiesConfiguration propertie
     CommonsConfigurationUtils.saveToFile(propertiesConfiguration, metadataFile);
   }
 
-  public static void updateMetadataProperties(SegmentMetadata segmentMetadata, Map<String, String> metadataProperties) {
+  public static SegmentMetadata updateMetadataProperties(SegmentMetadata segmentMetadata, Map<String,
```

Comment:
segmentMetadata is redundant here, right?  Can we can fetch it from segmentDirectory?

Reply:
ooh good point, thanks! fixed it